### DISCUSSION TITLE
Mirth Connect 4.0.1b293 service wouldn't start

### DISCUSSION THREAD

**Discussion by Farman09** [nextgenhealthcare/connect#5160] [General]
I have recently downloaded the latest version of Mirth Connect 4.0.1b293 Every time I try starting the Service from Mirth Connect Server Manager it throws me an error "Mirth Connect Service could not start".
Secondly, If I try to start launching the Administrator Launcher, it gives the following error. I have used the default 8080 and 8443 ports.

Unable to retrieve payload from HTTP request. URI: https://localhost:8443
java.lang.Exception: Unable to retrieve payload from HTTP request. URI: https://localhost:8443
	at com.mirth.connect.client.launcher.f.a(SourceFile:791)
	at com.mirth.connect.client.launcher.f.a(SourceFile:671)
	at com.mirth.connect.client.launcher.f.a(SourceFile:167)
	at com.mirth.connect.client.launcher.MirthClientLauncher.run(SourceFile:1230)
	at java.lang.Thread.run(Unknown Source)
Caused by: org.apache.http.conn.HttpHostConnectException: Connect to localhost:8443 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused: connect
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:156)
	at org.apache.http.impl.conn.BasicHttpClientConnectionManager.connect(BasicHttpClientConnectionManager.java:313)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at com.mirth.connect.client.launcher.f.a(SourceFile:775)
	... 4 more
Caused by: java.net.ConnectException: Connection refused: connect
	at java.net.DualStackPlainSocketImpl.waitForConnect(Native Method)
	at java.net.DualStackPlainSocketImpl.socketConnect(Unknown Source)
	at java.net.AbstractPlainSocketImpl.doConnect(Unknown Source)
	at java.net.AbstractPlainSocketImpl.connectToAddress(Unknown Source)
	at java.net.AbstractPlainSocketImpl.connect(Unknown Source)
	at java.net.PlainSocketImpl.connect(Unknown Source)
	at java.net.SocksSocketImpl.connect(Unknown Source)
	at java.net.Socket.connect(Unknown Source)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:368)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	... 13 more

**Comment by Farman09** [comment]
I have JAVA version 8 on Windows (10 x64 bit) operating system.

**Reply by pacmano1** [reply]
What JRE/JDK exactly?  OpenJDK?  Oracle?  Other?

**Reply by Farman09** [reply]
I have it from Oracle (jdk-8u331-windows-x64.exe)

**Comment by jonbartels** [comment]
Please use code formatting when posting error messages, it makes them easier to read - https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code . You can also edit posts to tweak formatting.

Java 8 on Windows should be well supported.

The error posted shows the error from the Launcher or Client - it cannot connect to the MC server, so you need to look at the MC server. Check the following:
* Can you reach https://localhost:8443/ with a browser or `curl` or `wget` ? Since its localhost the MC server probably isnt running
* MC `server.log` any errors there?
* Is the MC server process running? Is it listening on ports 8080 and 8443?
* Did you change the default ports (look at `mirth.properties` to see this)
* If you're running MC as a background service, try launching it with the `mcserver` executable. This is not suitable for normal use but it will spit out information to the command line if there is a problem that the background service isn't logging or showing well
* Any nag screens from Windows security tools or firewalls? Again its localhost so it should be fine, but a firewall or security tool could be blocking traffic

**Comment by Farman09** [comment]
* Gives error
* server.log is empty
* Nothing listening on ports 8080 and 8443
* I have added new rules for both ports to allow traffic on windows firewall
Still no luck

**Comment by jonbartels** [comment]
Ok, the next step is - Try launching it with the `mcserver` executable. 

```
cd /where/MC/was/installed
./mcserver
```
You should see output to the command line. Some common errors might be Java errors from a missing `JAVA_HOME` environment variable.

**Reply by Farman09** [reply]
It opens a new command prompt window and closes without any action. The issue still remains the same

**Reply by pacmano1** [reply]
open a command prompt and run

`java -version`

Do you get something like the below (I know the below is not windows, it is just the fact it reports a JVM is installed)

![image](https://user-images.githubusercontent.com/44065187/165798670-0e7476f5-79e8-46a4-ad33-c6268e1ea119.png)

**Reply by Farman09** [reply]
![image](https://user-images.githubusercontent.com/54945320/165799020-0a728b2f-2617-42fe-b470-048fc15d57cb.png)

**Reply by pacmano1** [reply]
That is java version 18, not 8.   Please remove it and install a lower version.  11 seems to always work fine.

**Reply by jonbartels** [reply]
https://s3.amazonaws.com/downloads.mirthcorp.com/connect-user-guide/latest/mirth-connect-user-guide.pdf Page 34 has relevant information that is relevant. HTH

**Reply by Farman09** [reply]
AWESOME! That worked!!!
I'm having version 8, but the SE Runtime Environment was the latest along with the old build. I just uninstalled the SE Runtime Enviornment (Build 18) it worked. 
I'm In

Thank you so much!

**Reply by siddharthsharma9258** [reply]
So overall java 11 will work. Mine is 

java version "17.0.2" 2022-01-18 LTS
Java(TM) SE Runtime Environment (build 17.0.2+8-LTS-86)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.2+8-LTS-86, mixed mode, sharing)

showing same symptom as OP.

**Reply by Lew3-cpu** [reply]
Hello , I am having the same issue and I have Java 8 installed. I have changed port to 8081 and still unable to launch.


Unable to retrieve payload from HTTP request. URI: http://localhost:8081/webstart
java.lang.Exception: Unable to retrieve payload from HTTP request. URI: http://localhost:8081/webstart
	at com.mirth.connect.client.launcher.f.a(SourceFile:788)
	at com.mirth.connect.client.launcher.f.a(SourceFile:668)
	at com.mirth.connect.client.launcher.f.a(SourceFile:168)
	at com.mirth.connect.client.launcher.MirthClientLauncher.run(SourceFile:1326)
	at java.base/java.lang.Thread.run(Unknown Source)

**Reply by pacmano1** [reply]
> Hello , I am having the same issue and I have Java 8 installed. I have changed port to 8081 and still unable to launch.
> 
> Unable to retrieve payload from HTTP request. URI: http://localhost:8081/webstart java.lang.Exception: Unable to retrieve payload from HTTP request. URI: http://localhost:8081/webstart at com.mirth.connect.client.launcher.f.a(SourceFile:788) at com.mirth.connect.client.launcher.f.a(SourceFile:668) at com.mirth.connect.client.launcher.f.a(SourceFile:168) at com.mirth.connect.client.launcher.MirthClientLauncher.run(SourceFile:1326) at java.base/java.lang.Thread.run(Unknown Source)

What is ine server's mirth.log file?

**Reply by Lew3-cpu** [reply]
ERROR 2025-03-10 13:34:26.178 [Main Server Thread] com.mirth.connect.server.Mirth: http.port port is already in use: 8081
ERROR 2025-03-10 13:34:26.185 [Main Server Thread] com.mirth.connect.server.Mirth: https.port port is already in use: 8445

**Reply by Lew3-cpu** [reply]
I don't know how to select a free port

**Reply by pacmano1** [reply]
- Make sure the server process is fully stopped.  use task manager (windows), or ps (linux) to confirm no mirth server java process is running
- Restart it.
- If you get the same error again it does mean some other process is taking those ports.
- In very rare instances Windows has used all ephermal ports for other tinhgs. 
- `netstat -ano` will tell you all ports in use.

**Reply by Lew3-cpu** [reply]
Proto  Local Address          Foreign Address        State           PID
  TCP    0.0.0.0:80             0.0.0.0:0              LISTENING       4

**Reply by Lew3-cpu** [reply]
from my understanding does this mean local port 8080 is in use by PID 4 ?

**Reply by pacmano1** [reply]
Not sure what you posted there.  The output of that netstat command would be gigantic.

**Reply by Lew3-cpu** [reply]
it is I thought that was port 8080

**Reply by Lew3-cpu** [reply]
if windows has used these ports is it just a matter of identifying a free port and changing it in the config?

**Reply by pacmano1** [reply]
That says 80, not 8080.  So perhaps you have IIS running.
```
TCP 0.0.0.0:80 0.0.0.0:0 LISTENING 4
```

**Reply by Lew3-cpu** [reply]
Okay, so from the cmd result port 8080 is free

**Reply by pacmano1** [reply]
yes, but you need 8080 and 8443 availabe. (although I think you can acutally edit mirth.properties and have no 8080 listener at all).

**Reply by Lew3-cpu** [reply]
@pacmano1  would you be willing to join a call to help resolve please?

**Reply by pacmano1** [reply]
If you want to engage our team professionaly (for a fee), please go to https://diridium.com/ and complete the Contact Us form.

We do help many people for free here, but we don't get on calls / web meetings without a professional engagement in place.

**Reply by Lew3-cpu** [reply]
Thank you I will take a look I feel I would benefit from a call more to learn
